FileIO/Legacy/MeshIO.cpp

```diff
@@ -103,7 +105,9 @@ MeshLib::Mesh* MeshIO::loadMeshFromFile(const std::string& file_name)
 				for (unsigned i = 0; i < nElements; ++i)
 				{
 					getline(in, line_string);
-					elements.push_back(readElement(line_string, nodes));
+					std::stringstream ss(line_string);
+					materials.push_back(readMaterialID(ss));
+					elements.push_back(readElement(ss, nodes));
```

Comment:
as i mentioned in my comments there are multiple ways to break this read-method. however, in my opinion we an leave it as it is for the most part as it is "only" a legacy interface and ogs5-read methods are similarly buggy (i.e. it is unlikely that someone comes up with a msh-file that will read in ogs5 but not in ogs6).

that said, in this particular case i would also *not* insert a nullptr but instead either abort reading the mesh or simply skip the element completely. in both cases we can guarantee consistent behaviour afterwards, whereas a nullptr within the element-vector will result in a crash later on...